apps/core/utils.py

```diff
@@ -59,3 +60,31 @@ def get_user_change_perms_for_status(request, obj):
         if user_level < right_to_change:
             return False
     return True
+
+
+def get_app_list(self, request):
+    admin_site_order = settings.ADMIN_SITE_ORDER
+    app_dict = self._build_app_dict(request)
+
+    app_list = sorted(app_dict.values(), key=lambda x: x["name"].lower())
+
+    for app in app_list:
+        if app["name"] in admin_site_order:
+            ordered_models = []
+            models = app["models"]
+            for model_name in admin_site_order[app["name"]]:
+                index = -1
+                for i, model in enumerate(models):
+                    if model["name"] == model_name:
+                        index = i
+                        break
+                if index == -1:
+                    raise ValueError("Ошибка в описании порядка моделей в админке.")
```

Comment:
Кажется, так код выглядит проще
```suggestion
                index = next((index for index, model in enumerate(models) if model['name'] == model_name), None)
                if index is not None:
                    ordered_models.append(models.pop(index))
                else:
                    raise ValueError("Ошибка в описании порядка моделей в админке.")
```

Reply:
Классная конструкция, буду знать!